src/app/input/input.component.spec.ts

```diff
@@ -43,13 +43,13 @@ describe('InputComponent', () => {
     await fixture.whenStable();
     const inputHarness = await loader.getHarness(MatInputHarness.with(
       {selector: '.inputTextBox'}));
-    component.data$.subscribe(
+    component.service.data$.subscribe(
       data => expect(data['inputString']).toEqual("")
     );
 
     await inputHarness.setValue("Some input text");
 
-    component.data$.subscribe(
+    component.service.data$.subscribe(
```

Comment:
Use injection to pass in the service, instead of referencing it directly from the component. (Don't component.service) Provide a service for a test using TestBed https://angular.io/guide/testing-services#angular-testbed

Reply:
change to "service.data$"